storage/replica_command.go

```diff
@@ -264,18 +311,14 @@ func (r *Replica) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, h ro
 
 	key := keys.TransactionKey(h.Txn.Key, h.Txn.ID)
 
-	// Fetch existing transaction if possible.
+	// Fetch existing transaction.
 	reply.Txn = &roachpb.Transaction{}
 	ok, err := engine.MVCCGetProto(batch, key, roachpb.ZeroTimestamp, true, nil, reply.Txn)
+	log.Infof("fetching on end txn %s on store %d: %t", uuid.UUID(h.Txn.ID).Short(), r.rm.StoreID(), ok)
 	if err != nil {
 		return reply, nil, err
-	}
-
-	if !ok {
-		// The transaction doesn't exist yet on disk; use the supplied version.
-		// The cloning shouldn't be strictly necessary since it's only passed
-		// to error constructors which itself make a copy, but let's be safe.
-		reply.Txn = h.Txn.Clone()
+	} else if !ok {
+		return reply, nil, util.Errorf("transaction does not exist: %s on store %d", h.Txn, r.rm.StoreID())
```

Comment:
should we have a new error type for this? After all, you'll return it on push, commit, heartbeat whenever you don't find an entry.